imap/imapd.c

```diff
@@ -2580,6 +2583,26 @@ static void cmdloop(void)
                     eatline(imapd_in, c);
                 }
             }
+            else if (!strcmp(cmd.s, "Uidbatches")) {
+                uint32_t size, low = 1, high = UINT32_MAX;
+
+                if (!imapd_index && !backend_current) goto nomailbox;
+
+                if (c != ' ') goto missingargs;
+                c = getuint32(imapd_in, &size);
+                if (c <= EOF) goto missingargs;
+                if (c == ' ' ) {
+                    c = getuint32(imapd_in, &low);
+                    if (c != ':') goto badsequence;
+                    c = getuint32(imapd_in, &high);
+                    if (c <= EOF) goto badsequence;
+                }
+                if (!IS_EOL(c, imapd_in)) goto extraargs;
+
+                cmd_uidbatches(tag.s, size, low, high);
```

Comment:
What if `high < low`, do we need to detect that and `goto badsequence`?  Or accept it and swap them around?  Or just produce no results?

Whatever we're doing, it'd be nice to see a test exercising it.

Reply:
Good question.  I've asked this question of the author and on the MAILMAINT list.  I will update the code when we have consensus on the correct answer